plugins/modules/netbox_power_outlet.py

```diff
@@ -0,0 +1,319 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# © 2020 Nokia
+# Licensed under the GNU General Public License v3.0 only
+# SPDX-License-Identifier: GPL-3.0-only
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: netbox_power_outlet
+short_description: Create, update or delete power outlets within Netbox
+description:
+  - Creates, updates or removes power outlets from Netbox
+notes:
+  - Tags should be defined as a YAML list
+  - This should be ran with connection C(local) and hosts C(localhost)
+author:
+  - Tobias Groß (@toerb)
+requirements:
+  - pynetbox
+version_added: '0.2.2'
+options:
+  netbox_url:
+    description:
+      - URL of the Netbox instance resolvable by Ansible control host
+    required: true
+    type: str
+  netbox_token:
+    description:
+      - The token created within Netbox to authorize API access
+    required: true
+    type: str
+  data:
+    type: dict
+    required: true
+    description:
+      - Defines the power outlet configuration
+    suboptions:
+      device:
+        description:
+          - The device the power outlet is attached to
+        required: true
+        type: raw
+      name:
+        description:
+          - The name of the power outlet
+        required: true
+        type: str
+      type:
+        description:
+          - The type of the power outlet
+        choices:
```

Comment:
This may be a good option to just set to `raw` and not track the options as it may change, but totally up to you at this point. I love the documentation of this aspect, but the maintenance can be burdensome when new ones are added for `type`.

Reply:
I have seen both options in other modules and I decided to go with the explicit documentation. From my point of view this is nicer to read for users of this ansible collection.